cmd/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+type Server struct {
+	mux *chi.Mux
+}
+
+func main() {
+	srv := Server{
+		mux: chi.NewRouter(),
+	}
+	srv.mux.Use(middleware.Logger)
+	srv.mux.Use(middleware.Recoverer)
+	srv.mux.Get("/hello-world", HelloWorldHandler)
+
+	srv.mux.Post("/todo", TodoHandler)
+
+	srv.mux.Get("/healthz", HealthzHandler)
+
+	log.Fatal(http.ListenAndServe(":8080", srv.mux))
+}
+
+func TodoHandler(w http.ResponseWriter, r *http.Request) {
+	b, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "Error reading request body")
+	}
+	fmt.Printf("Received todo: %s\n", b)
+	io.WriteString(w, "Received todo")
```

Comment:
It's likely your request body will be json, the package to convert from bytes to json is "json" and method "Unmarshal", so "json.Unmarshal" - check it out